s3.tf

```diff
@@ -74,6 +74,42 @@ resource "aws_s3_bucket" "apiary_data_bucket" {
   }
 }
 
+resource "aws_s3_bucket_versioning" "apiary_data_bucket_versioning" {
+  for_each = {
+    for schema in local.schemas_info : "${schema["schema_name"]}" => schema
+  }
+  bucket = each.value["data_bucket"]
+  versioning_configuration {
+    status = lookup(each.value, "s3_versioning_enabled", "Disabled")
+  }
+}
+
+resource "aws_s3_bucket_lifecycle_configuration" "apiary_data_bucket_versioning_lifecycle" {
+  for_each = {
+    for schema in local.schemas_info : "${schema["schema_name"]}" => schema
+  }
+  bucket = each.value["data_bucket"]
+  # Rule enabled when expiration max days is set
+  rule {
+    id     = "expire-noncurrent-versions-days"
+    status = lookup(each.value, "s3_versioning_expiration_days", "") != "" && lookup(each.value, "s3_versioning_max_versions_retained", "") == "" ? "Enabled" : "Disabled"
```

Comment:
This doesn't take the default value into account at all does it, the default is ""? So it will only be enabled or disabled depending on what is in the map, and the default value in the var is ignored here. But then the default value IS pulled in in the actual days or versions setting below. I'm not sure I understand... if we override it in the map, then the default will be ignored in both places. If we don't override it in the map, then it will be disabled. There is no case in which the default value will be used?

I guess in the second rule for max versions and days it could be used but only for setting the expiration_days, which only can happen if expiration_days is not set in the map, then it will use the default for it?

I wonder if it is simpler to just leave out the max versions rule completely. I'm not sure we have a case for it anyway. It's an option in the API but I don't think we want to use it.

Reply:
Okay, let's do it in that case only days